### ISSUE TITLE
HMR no longer works in target: node-webkit

### ISSUE THREAD

**Issue by lynxtaa** [webpack/webpack#6877]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
HMR fails to apply hot-update, when compiled for "node-webkit" target: `Uncaught ReferenceError: webpackHotUpdate is not defined.`:

![screenshot 2018-03-27 14 11 20](https://user-images.githubusercontent.com/22496946/37964093-f875d64a-31c8-11e8-8165-74aa4af256ef.png)

**If the current behavior is a bug, please provide the steps to reproduce.**

webpack configuration:
```javascript
module.exports = {
	mode: 'development',
	devtool: 'cheap-module-eval-source-map',
	target: 'node-webkit',
	entry: ['./src/index.js'],
	output: {
		path: join(__dirname, 'dist'),
		filename: 'bundle.js',
		publicPath: '/',
		devtoolModuleFilenameTemplate: 'webpack:///[absolute-resource-path]',
	},
	devServer: { hotOnly: true, publicPath: '/' },
	module: {
		rules: [
			{
				test: /\.jsx?$/,
				include: join(__dirname, 'src'),
				loader: 'babel-loader',
				options: { cacheDirectory: true, presets: ['react'] },
			},
		],
	},
	plugins: [
		new webpack.DefinePlugin({
			'process.env.NODE_ENV': JSON.stringify('development'),
		}),

		new webpack.HotModuleReplacementPlugin(),
	],
}
```
Reproduceable repo: https://github.com/lynxtaa/nw-hmr-fail

I'm using webpack-dev-server, but encountered the same problem with webpack-serve and webpack-hot-client, so I suspect that there is a problem with HotModuleReplacementPlugin. Webpack v3 works well.

**What is the expected behavior?**

HRM applying successfully in NW.js

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

NW.js 0.29.2, webpack 4.3.0, webpack-dev-server 3.1.1, Windows 10 x64

**Comment by lynxtaa** [comment]
The reason for this behavior is in `lib/WebpackOptionsDefaulter`:

```js
this.set("output.globalObject", "make", options => {
	switch (options.target) {
		case "web":
		case "electron-renderer":
			return "window";
		case "webworker":
			return "self";
		case "node":
		case "async-node":
		case "node-webkit":
		case "electron-main":
			return "global";
		default:
			return "self";
	}
});
```
If I change "target" to "electron-renderer" HMR works perfectly
```

**Comment by lynxtaa** [comment]
Fixed in v4.6.0
